tree-sitter-darklang/test/corpus/exhaustive/exprs/let_expr.txt

```diff
@@ -0,0 +1,170 @@
+==================
+simple let_expression
+==================
+
+let x = 1L
+x
+
+---
+
+(source_file
+  (expression
+    (let_expression
+      (keyword) (variable_identifier) (symbol)
+      (expression (simple_expression (int64_literal (digits (positive_digits)) (symbol))))
+      (expression (simple_expression (variable_identifier)))
+    )
+  )
+)
+
+
+==================
+simple let_expression -with indent
+==================
+
+let x =
+  1L
+x
+
+---
+
+(source_file
+  (expression
+    (let_expression
+      (keyword) (variable_identifier) (symbol)
+      (indent)
+      (expression (simple_expression (int64_literal (digits (positive_digits)) (symbol))))
+      (dedent)
+      (expression (simple_expression (variable_identifier)))
+    )
+  )
+)
+
+
+
+==================
+let expression with pipes -inline
+==================
+
+let x = 1L |> Stdlib.Option.Option.Some
+x
+
+---
+
+(source_file
+  (expression
+    (let_expression
+      (keyword) (variable_identifier) (symbol)
+      (expression
+        (pipe_expression
+          (expression (simple_expression (int64_literal (digits (positive_digits)) (symbol))))
+          (pipe_exprs
+            (symbol)
+            (pipe_expr
+              (pipe_enum
+                (qualified_type_name (module_identifier) (symbol) (module_identifier) (symbol) (type_identifier))
+                (symbol) (enum_case_identifier)
+              )
+            )
+          )
+        )
+      )
+      (expression (simple_expression (variable_identifier)))
+    )
+  )
+)
+
+
+==================
+let expression with one pipe -indented
+==================
+
+let x =
+  1L
+  |> Stdlib.Option.Option.Some
+x
+
+---
+
+(source_file
+  (expression
+    (let_expression
+      (keyword) (variable_identifier) (symbol)
+      (indent)
+      (expression
+        (pipe_expression
+          (expression (simple_expression (int64_literal (digits (positive_digits)) (symbol))))
+          (pipe_exprs
+            (symbol)
+            (pipe_expr
+              (pipe_enum
+                (qualified_type_name (module_identifier) (symbol) (module_identifier) (symbol) (type_identifier))
+                (symbol) (enum_case_identifier)
+              )
+            )
+          )
+        )
+      )
+      (dedent)
+      (expression (simple_expression (variable_identifier)))
+    )
+  )
+)
+
+
+==================
+let expression with multiple pipes -indented
+==================
+
+let wtTest =
+  testSource
+  |> LanguageTools.Parser.ParserTest.initialParse
+  |> Test.parseTest
+  |> Builtin.unwrap
+wtTest
```

Comment:
same here - the last line should be outside of the `let_expression`, I believe?

Reply:
I don't think so, unless I'm misunderstanding something.
here is our grammar before the changes : 
The field `body`(which I don't think is a good name for it) is included in the `let_expression` because we require an expression that follows `let something = something`

```
let_expression: $ =>
      seq(
        field("keyword_let", alias("let", $.keyword)),
        field("identifier", $.variable_identifier),
        field("symbol_equals", alias("=", $.symbol)),
        field("expr", $.expression),
        "\n",
        field("body", $.expression),
      ),
      ```

Reply:
you're right, I was confusing myself.